scripting/include/srccoop/playerpatch.inc

```diff
@@ -64,9 +64,29 @@ public void Hook_PlayerPreThinkPost(int iClient)
 	if(!g_pCoopManager.IsCoopModeEnabled())
 		return;
 	
+	static int iEntity = -1;
 	CBlackMesaPlayer pPlayer = CBlackMesaPlayer(iClient);
 	if(pPlayer.IsAlive())
 	{
+		if(IsValidEntity(pPlayer.GetCarriedObject().GetEntIndex()))
+		{
+			PreventWeaponFiring(pPlayer.GetEntIndex(), 1.0);
+		}
+		else if (GetEntProp(pPlayer.GetEntIndex(), Prop_Data, "m_nWaterLevel") == 3)
+		{
+			// Can *usually* assume that if you are parented and underwater that you are bound by a beneathticle
+			if (IsValidEntity(GetEntPropEnt(pPlayer.GetEntIndex(), Prop_Data, "m_hParent")))
```

Comment:
I didn't want to do class checks as well because this hook is fairly spammy, but it could check if the parent has `m_hPrey == pPlayer` as a lightweight check.